src/hotspot/share/classfile/placeholders.cpp

```diff
@@ -31,7 +31,26 @@
 #include "memory/resourceArea.hpp"
 #include "runtime/javaThread.hpp"
 #include "runtime/mutexLocker.hpp"
-#include "utilities/hashtable.inline.hpp"
+#include "utilities/resourceHash.hpp"
+
+class PlaceholderKey : public StackObj {
+  Symbol* _name;
+  ClassLoaderData* _loader_data;
+ public:
+  PlaceholderKey(Symbol* name, ClassLoaderData* l) : _name(name), _loader_data(l) {}
+
+  static bool equals(PlaceholderKey const& k1, PlaceholderKey const& k2) {
+    return (k1._name == k2._name && k1._loader_data == k2._loader_data);
+  }
+  static unsigned hash(PlaceholderKey const& k) {
+    return (unsigned) k._name->identity_hash();
```

Comment:
Should this be XORed with _loader_data?

Reply:
It wasn't before because it shared a hashcode with Dictionary, but now it doesn't anymore so I xor'ed it now.